### ISSUE TITLE
Artifacts or gray screen and stderr garbage for resolution <= 640x640 with AVC444

### ISSUE THREAD

**Issue by giox069** [FreeRDP/FreeRDP#5417]
Connecting to windows 10 or windows server 2012 with AVC444 and a remote resolution smaller or equal than 640x480 causes a gray screen and a lot of errors on stderr. It happened after [PR #5383](https://github.com/FreeRDP/FreeRDP/pull/5383)

`xfreerdp /v:192.168.56.2 /u:jjk001 /d:xxx /w:640 /h:480 /gfx-h264:AVC444`

the result is a gray background screen or some gray blocks, with a lot of errors on stderr.

In remmina we worked around this by calling
`freerdp_settings_set_uint32(rfi->settings, FreeRDP_GfxCapsFilter, 0xFFF); ` before connecting. But in this way we lost all RDPGFX_CAPVERSION_*

I opened this issue to leave it as a reference for the future, but...
what did we miss by disabling RDPGFX_CAPVERSION_* ? Performances? How much?

Cc: @antenore

**Comment by akallabeth** [comment]
There seems to be a bug in ms servers when newer caps are announced and certain resolution constraints are violated.
(Not exactly sure what they are, but classic 640x480 is one of those)

Started with caps >10.2
What you're missing:
Newer caps have some high dpi scaling options. Therefore currently you're not missing anything.

**Comment by giox069** [comment]
`freerdp_settings_set_uint32(rfi->settings, FreeRDP_GfxCapsFilter, 0xFFF);` seems to have a noticeable  performance loss. I can see it when opening the Windows 10 main start menu.
So, for now, the only solution I have found is to prevent the user to go under a minimum resolution when GFX is enabled (Remmina commit [here](https://gitlab.com/Remmina/Remmina/commit/87a34c421e640a3409e5ee7e8aaacbf6fc045fc0)).

**Comment by akallabeth** [comment]
@giox069 you can also try to enable a few caps (you disable all with that mask)

**Comment by giox069** [comment]
I have tried. For example, a mask of 0xE0 worked for the last windows 10 (still 1809) but not for Windows server 2012 R2. After 40 minutes wasted in testing... I gave up :smiley:
I noticed also that all stderror stuff comes from the h264 decoder, so I downloaded cisco's openh264 and compiled xfreerdp with ffmpeg=off and openh264=on. But I had the same problem: artifacts and  a lot of h264 decoder errors (different from the previous errors).
No problems, we will survive preventing the user to resize under 642x480.
If you will ever find a solution, just post on this issue :smiley:
